assets/src/edit-story/elements/image/display.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { PAGE_WIDTH, PAGE_HEIGHT } from '../../constants';
+import { useCanvas } from '../../components/canvas';
+import useDoubleClick from '../../utils/useDoubleClick';
+import { ElementWithPosition, ElementWithSize, ElementWithRotation } from '../shared';
+import { getImgProps, ImageWithScale } from './util';
+
+const Element = styled.div`
+	${ ElementWithPosition }
+	${ ElementWithSize }
+	${ ElementWithRotation }
+	overflow: hidden;
+`;
+
+const Img = styled.img`
+	position: relative;
+	${ ImageWithScale }
+`;
+
+function ImageDisplay( { id, src, origRatio, width, height, x, y, scale, offsetX, offsetY, rotationAngle, forwardedRef, onPointerDown } ) {
+	// Width and height are percent of the actual page dimensions,
+	// Thus 20-by-20 doesn't mean square, but "same as page ratio".
+	const actualRatio = width / height * PAGE_WIDTH / PAGE_HEIGHT;
```

Comment:
Just realized that currently, any element can be resized to `0`, this would break the logic, I assume?

Reply:
Division by zero might cause a black hole or similar singularity, yes. But let's fix that elsewhere.